src/main/java/seedu/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,73 @@
+package seedu.duke.command;
+
+import seedu.duke.data.UserData;
+import seedu.duke.event.Event;
+import seedu.duke.event.EventList;
+import seedu.duke.exception.DukeException;
+import seedu.duke.exception.WrongNumberFormatException;
+import seedu.duke.exception.WrongNumberOfArgumentsException;
+import seedu.duke.storage.Storage;
+import seedu.duke.ui.Ui;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+import static seedu.duke.parser.DateTimeParser.dateParser;
+
+public class DeleteCommand extends Command {
+    private String listType;
+
+    public DeleteCommand(String listType, String command) {
+        this.isExit = false;
+        this.listType = listType;
+        this.command = command;
+    }
+
+    public static Command parse(String input) throws WrongNumberOfArgumentsException {
+        String[] inputParameters = input.trim().split(" ", 3);
+
+        if (inputParameters.length < 2) {
+            throw new WrongNumberOfArgumentsException("Event type not provided.");
+        } else if (inputParameters.length < 3) {
+            throw new WrongNumberOfArgumentsException("Event index not provided.");
+        }
+
+        String listType = inputParameters[0];
+        String eventIndex = inputParameters[1].trim();
+
+        return new DeleteCommand(listType, eventIndex);
+    }
+
+    @Override
+    public void execute(UserData data, Ui ui, Storage storage) throws DukeException {
+        listType = capitaliseFirstLetter(listType);
+        EventList eventList = data.getEventList(listType);
+        String[] eventIndexArray = command.split(" ",2);
+
+        try {
+            int eventIndex = Integer.parseInt(eventIndexArray[0]) - 1;
```

Comment:
You can check this in the parse method to avoid this surrounding your code with this large try catch.

Reply:
Ok thanks for the feedback!